That's not good, I think:
```
error: builder for '/nix/store/914fhpz4gpfnf7b248lyh6xampaj9ac9-apfel-3.0.6.drv' failed with exit code 2;
       last 10 log lines:
       > f951: Warning: '-Werror=' argument '-Werror=format-security' is not valid for Fortran
       > /private/tmp/nix-build-apfel-3.0.6.drv-0/ccsnYzSU.s:80:15: error: index must be an integer in range [-256, 255].
       >         ldr     x4, [x4, ___stack_chk_guard];momd
       >                          ^
       > /private/tmp/nix-build-apfel-3.0.6.drv-0/ccsnYzSU.s:596:15: error: index must be an integer in range [-256, 255].
       >         ldr     x0, [x0, ___stack_chk_guard];momd
       >                          ^
       > make[1]: *** [Makefile:1255: FTDY/src/ComputeHardCrossSectionsDY.lo] Error 1
       > make[1]: Leaving directory '/private/tmp/nix-build-apfel-3.0.6.drv-0/source/src'
       > make: *** [Makefile:500: all-recursive] Error 1
       For full logs, run 'nix log /nix/store/914fhpz4gpfnf7b248lyh6xampaj9ac9-apfel-3.0.6.drv'.
```
It affects `apfel`, `python3Packages.primer3` and `python3Packages.word2vec` according to your log. I don't know if they're all Fortran-related, but the `f951` warning above suggests `apfel` is.

--- a/pkgs/build-support/cc-wrapper/default.nix
+++ b/pkgs/build-support/cc-wrapper/default.nix
@@ -245,12 +245,19 @@ stdenv.mkDerivation {
       wrap ${targetPrefix}gdc $wrapper $ccPath/${targetPrefix}gdc
     ''
 
-    + optionalString cc.langFortran or false ''
+    + optionalString cc.langFortran or false (''
       wrap ${targetPrefix}gfortran $wrapper $ccPath/${targetPrefix}gfortran
       ln -sv ${targetPrefix}gfortran $out/bin/${targetPrefix}g77
       ln -sv ${targetPrefix}gfortran $out/bin/${targetPrefix}f77
       export named_fc=${targetPrefix}gfortran
     ''
+    # Darwin aarch64 fortran compilations seem to fail otherwise, see:
+    # https://github.com/NixOS/nixpkgs/issues/140041
+    + (if (stdenvNoCC.isDarwin && stdenvNoCC.isAarch64) then ''
+      export fortran_hardening="pic strictoverflow relro bindnow"
+    '' else ''
+      export fortran_hardening="pic strictoverflow relro bindnow stackprotector"
+    ''))
 
     + optionalString cc.langJava or false ''
       wrap ${targetPrefix}gcj $wrapper $ccPath/${targetPrefix}gcj

--- a/pkgs/build-support/cc-wrapper/fortran-hook.sh
+++ b/pkgs/build-support/cc-wrapper/fortran-hook.sh
@@ -5,7 +5,7 @@ export FC${role_post}=@named_fc@
 
 # If unset, assume the default hardening flags.
 # These are different for fortran.
-: ${NIX_HARDENING_ENABLE="stackprotector pic strictoverflow relro bindnow"}
+: ${NIX_HARDENING_ENABLE="@fortran_hardening@"}
 export NIX_HARDENING_ENABLE
 
 unset -v role_post

